File: 프로그래머스/1/42748. K번째수/K번째수.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;
    for(int i=0;i<commands.size();i++){
        int s=commands[i][0];
        int e=commands[i][1];
        int k=commands[i][2];
        
        vector<int> sliced_array(array.begin()+(s-1),array.begin()+e);
        sort(sliced_array.begin(),sliced_array.end());
        answer.push_back(sliced_array[k-1]);
    }
    
    return answer;
}

File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.*;

class Solution {
    public int[] solution(int[] array, int[][] commands) {

        int[] answer = new int[commands.length];

        for (int i = 0; i < commands.length; i++) {

            int start = commands[i][0] - 1; // i
            int end = commands[i][1];       // j
            int k = commands[i][2] - 1;     // k

            int[] temp = Arrays.copyOfRange(
                array,
                start,
                end
            );

            Arrays.sort(temp);

            answer[i] = temp[k];
        }

        return answer;
    }
}